sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -1084,7 +1084,12 @@ export class MessageReceiver extends LinkEntity {
         if (options.propertiesToModify) params.message_annotations = options.propertiesToModify;
         delivery.modified(params);
       } else if (operation === DispositionType.deadletter) {
-        delivery.reject(options.error || {});
+        options.error = options.error || {};
```

Comment:
For clarity (unless it's intentional) it's probably better to declare a local variable here rather than updating `options`.